fhir-odoo/src/main/java/com/ozonehis/fhir/odoo/patient/impl/PatientServiceImpl.java

```
@@ -0,0 +1,79 @@
+import org.hl7.fhir.r4.model.Patient;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Slf4j
+@Service
+@SuppressWarnings("rawtypes, unchecked")
+public class PatientServiceImpl implements PatientService {
+
+    private final CountryService countryService;
+
+    private final CountryStateService countryStateService;
+
+    private final PartnerService partnerService;
+
+    private final PartnerMapper partnerMapper;
+
+    @Autowired
+    public PatientServiceImpl(
+            CountryService countryService,
+            CountryStateService countryStateService,
+            PartnerService partnerService,
+            PartnerMapper partnerMapper) {
+        this.countryService = countryService;
+        this.countryStateService = countryStateService;
+        this.partnerService = partnerService;
+        this.partnerMapper = partnerMapper;
+    }
+
+    @Override
+    public Patient create(Patient patient) {
+        Map<String, Object> resourceMap = new HashMap<>();
+        resourceMap.put(OdooConstants.MODEL_FHIR_PATIENT, patient);
+
+        Optional<Country> country =
+                countryService.getByName(patient.getAddress().get(0).getCountry());
+        country.ifPresent(value -> resourceMap.put(OdooConstants.MODEL_COUNTRY, value));
+
+        Optional<CountryState> countryState =
+                countryStateService.getByName(patient.getAddress().get(0).getState());
+        countryState.ifPresent(value -> resourceMap.put(OdooConstants.MODEL_COUNTRY_STATE, value));
+
+        Partner partner = partnerMapper.toOdoo(resourceMap);
+
+        int id = partnerService.create(Map.of()); // TODO: Add logic to convert Partner to Map and pass
+        if (id != 0) {
+            log.info("Partner created in Odoo with id {}", id);
+        }
+        return patient;
+    }
+
+    @Override
+    public Optional<Patient> getById(@Nonnull String id) {
+        return null; // TODO: Implement
+    }
```

If you are not implementing this at this stage, throw `UnsupportedOperationException("Not yet implemented");` instead of returning null.